firmware/common/include/interface/blitter.h

```diff
@@ -14,14 +14,30 @@
 #define _BLITTER_H
 
 struct BlitterArea {
-	uint16_t 	address;
-	uint8_t 	page,_filler;
-	uint32_t  	stepSize;
-	uint32_t  	stepOffset;
-	uint32_t  	stepCount;
+	uint16_t	address;
+	uint8_t		page;
+	uint8_t		format;     // one of BLTFMT_*
+	uint8_t		k;			// mask in src, solid value in target.
+	uint8_t  	width;		// Size of line, in units (ignored on target).
+	uint16_t  	height;		// Number of lines (ignored on target).
+	int16_t		stride;		// Number of bytes between start of each line in memory.
 };
 
+// Format types for BlitterArea.format field
+#define BLTFMT_BYTE 0		// whole byte
+#define BLTFMT_PAIR 1		// 2 4-bit values, aka nibbles (src only)
+#define BLTFMT_BITS 2		// 8 1-bit values (src only)
+#define BLTFMT_HIGH 3		// High nibble (target only)
+#define BLTFMT_LOW  4		// Low nibble (target only)
+
+
 uint8_t BLTSimpleCopy(uint8_t pageFrom,uint16_t addressFrom, uint8_t pageTo, uint16_t addressTo, uint16_t transferSize);
+
+// Values for BLTComplexCopy action param
+#define BLTACT_COPY 0		// Straight rectangle copy
+#define BLTACT_MASK 1		// Copy, but only where src != srcarea.v (transparent value)
+#define BLTACT_SOLID 2		// Fill with destarea.k (solid value), but only where src != srcarea.k (transparent value)
+
```

Comment:
I think adding BLTACT_XOR 3 might be useful, where dest value is xored with current memory content.

Reply:
I can definitely see the appeal of a `target = target XOR src` operation, but didn't quite understand the other one. Did you mean xor the target with a constant value?

Reply:
There's no BASIC interface as yet.